runner.py

```diff
@@ -0,0 +1,173 @@
+        bytecode.TypeofInst(result, Var('x')))
+    env[SSym('inst/alloc')] = inst_function(
+        SSym('inst/alloc'), [Var('n')],
+        bytecode.AllocInst(result, Var('n')))
+    env[SSym('inst/load')] = inst_function(
+        SSym('inst/load'), [Var('v'), Var('n')],
+        bytecode.LoadInst(result, Var('v'), Var('n')))
+    env[SSym('inst/store')] = inst_function(
+        SSym('inst/store'), [Var('v'), Var('n'), Var('x')],
+        bytecode.StoreInst(Var('v'), Var('n'), Var('x')),
+        bytecode.ReturnInst(bytecode.NumLit(scheme.SNum(0))),
+        should_return=False)
+    env[SSym('inst/length')] = inst_function(
+        SSym('inst/length'), [Var('v')],
+        bytecode.LengthInst(result, Var('v')))
+    env[SSym('inst/pointer=')] = inst_function(
+        SSym('inst/pointer='), [Var('a'), Var('b')],
+        bytecode.BinopInst(result, Binop.PTR_EQ, Var('a'), Var('b')))
+    env[SSym('inst/number=')] = inst_function(
+        SSym('inst/number='), [Var('a'), Var('b')],
+        bytecode.BinopInst(result, Binop.NUM_EQ, Var('a'), Var('b')))
+    env[SSym('inst/symbol=')] = inst_function(
+        SSym('inst/symbol='), [Var('a'), Var('b')],
+        bytecode.BinopInst(result, Binop.SYM_EQ, Var('a'), Var('b')))
+    env[SSym('inst/number<')] = inst_function(
+        SSym('inst/number<'), [Var('a'), Var('b')],
+        bytecode.BinopInst(result, Binop.NUM_LT, Var('a'), Var('b')))
+    env[SSym('inst/trap')] = inst_function(
+        SSym('inst/trap'), [],
+        bytecode.TrapInst("(trap)"),
+        should_return=False)
+
+
+def add_builtins(env: Dict[SSym, Value]) -> None:
+    """Add builtins to the environment."""
+    add_intrinsics(env)  # @TODO: Don't do this, for greater flexibility?
+    code = scheme.parse("""
+    (define (trap) (inst/trap))
+    (define (assert b) (if b 0 (trap)))
+    (define (typeof x) (inst/typeof x))
+
+    (define (pointer= a b) (inst/pointer= a b))
+    (define (symbol= a b)
+      (assert (inst/symbol= (typeof a) 'symbol))
+      (assert (inst/symbol= (typeof b) 'symbol))
+      (inst/symbol= a b))
+    (define (number= a b)
+      (assert (symbol= (typeof a) 'number)
+      (assert (symbol= (typeof b) 'number)
+      (inst/number= a b))))
+
+    (define (number< a b)
+      (assert (symbol= (typeof a) 'number))
+      (assert (symbol= (typeof b) 'number))
+      (inst/number< a b))
+
+    (define (vector-length v)
+      (assert (symbol= (typeof v) 'vector))
+      (inst/length v))
+
+    (define (vector-index v n)
+      (assert (symbol= (typeof v) 'vector))
+      (assert (symbol= (typeof n) 'number))
+      (assert (number< -1 n))
+      (assert (number< n (vector-length v)))
+      (inst/load v n))
+
+    (define (vector-set! v n x)
+      (assert (symbol= (typeof v) 'vector))
+      (assert (symbol= (typeof n) 'number))
+      (assert (number< -1 n))
+      (assert (number< n (vector-length v)))
+      (inst/store v n x))
+
+    (define (vector-make/recur len idx v x)
```

Comment:
Could you add a a comment/example of what this does? (same for `vector=/recur` later on)

Reply:
It's just needed to build the loop inside the function, since they need parameters than we expose. It's not intended to be used (in a more capable language I would've had it nested inside).

Reply:
Also… what do you mean by this? I can add comments in the source code if we merge #37 first.